skbio/sequence/_sequence.py

```diff
@@ -1408,6 +1408,47 @@ def count(self, subsequence, start=None, end=None):
         return self._string.count(
             self._munge_to_bytestring(subsequence, "count"), start, end)
 
+    @experimental(as_of="0.4.2-dev")
+    def replace(self, where, character):
+        """Replace values in this sequence with a different character.
+
+        Parameters
+        ----------
+        where: boolean vector
```

Comment:
Since you're converting `where` into an array of positions with `_munge_to_index_array`, this can accept (in addition to boolean vectors) an iterable of indices (including slices) or a string that is a key in positional metadata pointing to a boolean vector.

Change to:

```
where : 1D array_like (bool) or iterable (slices or ints) or str
    Indicates positions in the sequence to replace with `character`. Can be a boolean vector, an iterable of indices/slices, or a string that is a key in `positional_metadata` pointing to a boolean vector.
```